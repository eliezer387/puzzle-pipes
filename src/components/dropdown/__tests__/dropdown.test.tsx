import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Dropdown from "..";

const setup = () => {
  const options = [
    { label: "Option 1", value: "1" },
    { label: "Option 2", value: "2" },
  ];
  const onChange = jest.fn();
  const utils = render(<Dropdown options={options} change={onChange} />);
  const dropdown = utils.getByDisplayValue("Option 1");
  return {
    dropdown,
    utils,
  };
};

describe("Dropdown", () => {
  it("should render", () => {
    const { dropdown } = setup();
    expect(dropdown).toBeDefined();
  });

  it("should match snapshot", () => {
    const { dropdown } = setup();
    expect(dropdown).toMatchSnapshot();
  });

  it("should call onChange", () => {
    const { dropdown, utils } = setup();
    fireEvent.change(dropdown, { value: "1" });
    expect(utils.getByDisplayValue("Option 1")).toBeDefined();
  });
});
