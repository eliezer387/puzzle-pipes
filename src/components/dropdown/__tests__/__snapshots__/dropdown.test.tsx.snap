// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match snapshot 1`] = `
<select
  name="dropdown"
>
  <option
    value="1"
  >
    Option 1
  </option>
  <option
    value="2"
  >
    Option 2
  </option>
</select>
`;
